public/app/percona/dbaas/components/DBaasRouting/DBaaSRouting.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { FC, useEffect, useRef, useMemo } from 'react';
+import { getKubernetes as getKubernetesSelector } from '../../../shared/core/selectors';
+import { fetchKubernetesAction } from '../../../shared/core/reducers';
+import { CHECK_OPERATOR_UPDATE_CANCEL_TOKEN, GET_KUBERNETES_CANCEL_TOKEN } from '../Kubernetes/Kubernetes.constants';
+import { useCancelToken } from '../../../shared/components/hooks/cancelToken.hook';
+import { useDispatch, useSelector } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import { Spinner, useStyles } from '@grafana/ui/src';
+import { getStyles } from '../DBaasRouting/DBaasRouting.styles';
+
+export const DBaaSRouting: FC<{}> = ({}) => {
+  const styles = useStyles(getStyles);
+  const { result: kubernetes = [], loading: kubernetesLoading } = useSelector(getKubernetesSelector);
+  const firstRender = useRef(true);
+  const [generateToken] = useCancelToken();
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (firstRender.current) {
```

Comment:
why do we need this flag?

Reply:
Thanks! Lately, for some reason I've been overusing ref. I thought and find out that I can just simply by removing the default value of the array from kubernetes and check it

Reply:
So I'll fix it

Reply:
> You have to wait for the loading to vanish before trying assertions:

it works for me now... but the problem was that i didn't wrap my component in tests with <Router/> now it works